src/main/java/org/roaringbitmap/longlong/RoaringIntPacking.java

```diff
@@ -0,0 +1,88 @@
+package org.roaringbitmap.longlong;
+
+import java.math.BigInteger;
+import java.util.Comparator;
+
+/**
+ * Used to hold the logic packing 2 integers in a long, and separating a long in two integers. It is
+ * useful in {@link RoaringTreeMap} as the implementation split the input long in two integers, one
+ * used as key of a NavigableMap while the other is added in a Bitmap
+ * 
+ * @author Benoit Lacelle
+ *
+ */
+// Hidden as it holds implementation details for RoaringTreeMap. We may decide to change the logic
+// here, hence it should
+// not be used elsewhere
+class RoaringIntPacking {
+
+  /**
+   * 
+   * @param id a long to decompose into two integers
+   * @return an int holding 32 bits of information of a long. Typically the highest order bits.
```

Comment:
The design issue might be related to the failing test with .add(0, 3L * Integer.MAX_VALUE). I need to focus on this